samples/snippets/create_cluster_on_gke.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This sample walks a user through creating a Cloud Dataproc cluster using
+# the Python client library.
+#
+# This script can be run on its own once you have replaced the variables in main():
+#   python create_cluster_on_gke.py
+
+
+# [START dataproc_create_cluster_on_gke]
+
+from google.cloud import dataproc_v1 as dataproc
+
+# from google.cloud import dataproc_v1.types as dataproc_types
+
+
+def main() -> None:
+    # Developer (TODO): You may find it helpful to create a Dict object with other needed pieces of information.
+    # Developer (TODO): Replace with your project_id.
+    project_id = "your-project-id"
+    # Developer (TODO): Replace with the region you are using.
+    region = "your-region"
+    # Developer (TODO): Replace with the region and zone you are using to use as a location for your GKE cluster.
+    gke_zone = "your-zone"
+    # Developer (TODO): Replace with the cluster_name of the Dataproc cluster on which you want to run a virtual cluster on GKE.
+    dp_cluster_name = "your-dp-cluster-name"
+    # Developer (TODO): Replace with gke_cluster_name for the virtual cluster you will run on GKE.
+    gke_cluster_name = "your-gke-cluster-name"
+    # Developer (TODO): Replace with your desired node_pool. If unspecified, Dataproc will use your default node_pool.
+    node_pool = "your-node-pool"
+    # Developer (TODO): Replace with your phs_cluster name if you plan to run a Spark job.
+    phs_cluster = "your-phs-cluster"
+    # Developer (TODO): Replace with your gcs_bucket_name for staging.
+    bucket = "your-staging-bucket"
```

Comment:
I would suggest making this whole sample a single function and passing these in where appropriate as function args. The explanation for TODOs can be moved into a doc string, [for example](https://github.com/googleapis/python-dataproc/blob/eecda18d1a6784cdbd213240692c1a67c9165a54/samples/snippets/create_cluster.py#L31).